src/layer/AbstractSentinelHubV3Layer.ts

```diff
@@ -236,7 +245,7 @@ export class AbstractSentinelHubV3Layer extends AbstractLayer {
   }
 
   public supportsApiType(api: ApiType): boolean {
-    if (this.dataProduct && !SUPPORTED_DATA_PRODUCTS_PROCESSING.includes(this.dataProduct)) {
+    if (!this.evalscript || SUPPORTED_DATA_PRODUCTS_PROCESSING.includes(this.dataProduct)) {
```

Comment:
Sorry for so many complications in this MR

I checked and the combination of the evalscript being successfully fetched and set for the data products in `SUPPORTED_DATA_PRODUCTS_PROCESSING` in `fetchLayerParamsFromSHServiceV3` causes problems, because the evalscript is set and JS doesn't even check the condition after `||`, so it uses ProcessAPI. Additionally, even if the WMS ApiType is selected, it will still use the evalscript instead of the data product.

I realized that `dataProduct` can also be passed to ProcessAPI - and this is already done in `src/layer/processing.ts` in case the `dataProduct` parameter is defined and `evalscript` isn't. ([lines 156 to 161](https://github.com/sentinel-hub/sentinelhub-js/blob/master/src/layer/processing.ts#L156))

I tweaked the code a bit, so it works also for the data product in `SUPPORTED_DATA_PRODUCTS_PROCESSING`  (for me locally):

- changed the new `if` in `fetchLayerParamsFromSHServiceV3` so that it checks if the `dataProduct` is in `SUPPORTED_DATA_PRODUCTS_PROCESSING`, so the evalscript is not requested if it is

```js
if (!layerParams['evalscript'] && layerParams['dataProduct'] && !SUPPORTED_DATA_PRODUCTS_PROCESSING.includes(layerParams['dataProduct'])) {
  // call fetchDataProduct() ... 
}
```

- changed the checks in this function (`supportsApiType`) entirely

```js
  public supportsApiType(api: ApiType): boolean {
    if (this.dataProduct && SUPPORTED_DATA_PRODUCTS_PROCESSING.includes(this.dataProduct)){
      return api === ApiType.PROCESSING;
    }
    return api === ApiType.WMS || (api === ApiType.PROCESSING && !!this.dataset && !!this.evalscript);
  }
```

The changed tests can be changed back

Reply:
Wasn't aware of dataProduct input parameter from processing API. Thanks for such insights!